src/database/Database.cpp

```diff
@@ -335,6 +335,18 @@ Database::getUpsertTimer(std::string const& entityName)
         .TimeScope();
 }
 
+void
+Database::maybeArtificiallySleepForSimulatedLatency()
+{
+#ifdef BUILD_TESTS
```

Comment:
I think we can have this enabled in all builds, we just need to add `ARTIFICIAL_DATABASE_LATENCY_FOR_TESTING` to `TESTING_ONLY_OPTIONS` in config.cpp

Reply:
Ah, ok. Fixed.